tests/main/snapctl/task.yaml

```diff
@@ -0,0 +1,80 @@
+summary: Check that `snapctl` can be run from within hooks
+
+systems: [-ubuntu-core-16-64]
+
+prepare: |
+    echo "Ensure we have a core snap with snap run"
+    $SPREAD_PATH/tests/lib/reset.sh
+    snap install --channel=candidate ubuntu-core
+    snapbuild $TESTSLIB/snaps/snapctl-hooks .
+    snap install snapctl-hooks_1.0_all.snap
+
+    # We want to use the in-tree snap-exec and snapctl, not the ones in the core
+    # snap. To accomplish that, we'll just unpack the core we just grabbed,
+    # shove the new snap-exec and snapctl in there, and repack it.
+
+    # First of all, unmount the core
+    systemctl stop snapd.service
+    core="$(realpath /snap/ubuntu-core/current)"
+    snap="$(mount | grep "$core" | awk '{print $1}')"
+    umount "$core"
+
+    # Now unpack the core, inject the new snap-exec and snapctl into it, and
+    # repack it.
+    unsquashfs "$snap"
```

Comment:
I wonder if we should simply do that as part of the global prepare? ensure that the ubuntu-core snap has snapd/snap-exec from our build?